vignettes/including-data-in-teal-applications.Rmd

```diff
@@ -0,0 +1,251 @@
+Use the `datanames` function to modify the `datanames` property.
+
+```{r}
+data_with_objects <- teal_data(iris = iris, cars = mtcars)
+data_with_code <- teal_data() %>% 
+  within({
+    iris <- iris
+    cars <- mtcars
+    not_a_dataset <- "data source credits"
+  })
+datanames(data_with_objects)
+datanames(data_with_code)
+datanames(data_with_code) <- c("iris", "cars")
+datanames(data_with_code)
+```
+
+All `teal` modules take a `datanames` argument that determines which datasets they are to have access to.
+Only objects enumerated in the `datanames` property of the `teal_data` object can be used.
+
+For a detailed explanation of datanames, see [this `teal.data` vignette](https://insightsengineering.github.io/teal.data/latest-tag/articles/teal-data.html).
+
+(back to [General Data])
+
+##### `join_keys`
+
+Using relational data requires specifying joining keys for each pair of datasets.
+Primary keys are unique row identifiers in individual datasets and thus should be specified for each dataset.
+Foreign keys describe mapping of variables between datasets.
+Joining keys are stored in the `join_keys` property, which can be set when creating a `teal_data` object, using the `join_keys` argument, or using the `join_keys` function.
+
+```{r}
+ds1 <- data.frame(
+  id = seq(1, 10),
+  group = rep(c("A", "B"), each = 5)
+)
+ds2 <- data.frame(
+  group = c("A", "B"),
+  condition = c("condition1", "condition2")
+)
+keys <- join_keys(
+  join_key("DS1", keys = "id"),
+  join_key("DS2", keys = "group"),
+  join_key("DS1", "DS2", keys = c("group" = "group"))
+)
+data_relational1 <- teal_data(DS1 = ds1, DS2 = ds2, join_keys = keys)
+data_relational2 <- teal_data(DS1 = ds1, DS2 = ds2)
+join_keys(data_relational2) <- keys
+```
+
+For a detailed explanation of join keys, see [this `teal.data` vignette](https://insightsengineering.github.io/teal.data/latest-tag/articles/join-keys.html).
+
+(back to [`ADaM` Data])
+
+##### `verified`
+
+`teal_data` allows for tracking code from data creation through filtering through analysis so that the whole process can be reproduced.
+The `verified` property designates whether or not reproducibility has been confirmed.
+`teal_data` objects that are created empty and are only modified by evaluating code in them are verified by default.
+Ones created with data objects alone or with data objects and code are not verified by default but can become verified by running the `verify` function.
+
+```{r}
+data_with_code
+
+data_with_objects_and_code <- teal_data(iris = iris, cars = mtcars, code = expression(iris <- iris, cars <- mtcars))
+data_with_objects_and_code
+
+data_with_objects_and_code_verified <- verify(data_with_objects_and_code)
+data_with_objects_and_code_verified
+```
+
+
+For a detailed explanation of verification, see [this `teal.data` vignette](https://insightsengineering.github.io/teal.data/latest-tag/articles/teal-data.html).
```

Comment:
Let's refer to this https://github.com/insightsengineering/teal.data/blob/refactor/vignettes/teal-data-reproducibility.Rmd

```suggestion
For a detailed explanation of verification, see [the "`teal.data` reproducibility" vignette](https://insightsengineering.github.io/teal.data/latest-tag/articles/teal-data-reproducibility.html).
```